pandas/core/series.py

```diff
@@ -2010,7 +2010,7 @@ def map_f(values, f):
                 arg = self._constructor(arg, index=arg.keys())
 
             indexer = arg.index.get_indexer(values)
-            new_values = com.take_1d(arg.values, indexer)
+            new_values = com.take_1d(arg.get_values(), indexer)
```

Comment:
so the way to do this is actually
``arg.values.take_nd(indexer)`` which will return a ``Categorical``. 

so maybe make this mod in ``core.common.take_nd`` (which is aliases to ``take_1d``). if its a categorical_dtype, then short-circuit as I show above. (it ultimately calls ``take_1d`` as well but with the codes)